gpii/node_modules/contextManager/src/ContextManager.js

```diff
@@ -118,61 +120,45 @@ gpii.contextManager.onEnvironmentChanged = function (that, body) {
             // ms since UTC epoch, eg: 1421066816897
             "http://registry.gpii.net/common/environment/timestamp": date.getTime(),
             // time of day, eg: "18:30"
             "http://registry.gpii.net/common/environment/timeOfDay": date.getHours() + ":" + date.getMinutes()
         });
-    }, 20000); //report time every 20 seconds
+    }, 20000); // report time every 20 seconds
 };
 
 /**
  * Triggered on destruction of the context manager component. Stops the temporal reporters
  * time reports
  *
- * @that {Object} The context manager object
+ * @param that {Component} The context manager object
  **/
 gpii.contextManager.stopTemporalEnvironmentReporter = function (that) {
-    clearInterval(that.intervalID);
+    clearInterval(that.temporalIntervalID);
 };
 
 /**
- * This function is listening to the changeApplier of the contextManager ("currentContext" path)
+ * This function is listening to the changeApplier of the contextManager ("currentEnvironment" path)
  *
- * On changes to the currentContext part of the model it will:
- * 1) parse the context against the currently logged in users' conditions
+ * On changes to the currentEnvironment part of the model it will:
+ * 1) parse the context against the currently logged in user's conditions
  * 2) if the new calculated active context is different from the currently applied context,
- *    the lifecycle managers update functionality will be called and the applied context and
+ *    the LifecycleManager's update functionality will be called and the applied context and
  *    configuration will be stored in the active session of the lifecycle manager.
  *
- * @that {Object} The context manager object
- * @lifecycleManager {Object} the lifecycleManager component from which the current
+ * @param that {Object} The context manager object
+ * @param lifecycleManager {Component} The lifecycleManager component from which the current
  *    session will be retrieved
- * @context {Object} The model as reporterd by the changeApplier (i.e. the content of
- *    that.model.currentContext)
-
-    var fullPayload = fluid.extend(true, {}, activeSession.model);
-    var oldActiveContext = fullPayload.activeContextName;
-
-    // find and update context:
-    that.addActiveContext(fullPayload);
-
-    if (oldActiveContext === fullPayload.activeContextName) {
-        fluid.log("contextManager: Same context as before (" + oldActiveContext + ") so doing nothing");
-        return;
-    }
-    fluid.log("contextManager: New active contexts: " + fullPayload.activeContextName);
-
-    lifecycleManager.addLifecycleInstructions(fullPayload);
-    var response = lifecycleManager.update(fullPayload);
+ * @param forceContextName {String} [optional] A context name to be unconditionally selected
  **/
-gpii.contextManager.evaluateConditions = function (that, lifecycleManager) {
+gpii.contextManager.contextChanged = function (that, lifecycleManager, forceContextName) {
```

Comment:
Do you think it makes sense to add a test for checking the behavior of this function when `forceContextName` is specified as an argument?

Reply:
Due to the fact that the function results in calls to the lifecycle manager (and subsequent calls to settingshandlers, etc), I've put the test for this functionality in the PCPIntegration test: https://github.com/GPII/universal/pull/561/files#diff-cfe024db61e395e38877b08299e4ed67R398